tdishr/TdiCompile.c

```diff
@@ -60,7 +60,7 @@ extern int TdiYacc();
 	supports (`) on COMPILE
 */
 
-static void cleanup_compile(ThreadStatic * TdiThreadStatic_p){
+static void cleanup_compile(TDITHREADSTATIC_ARG){
```

Comment:
Why this macro.  I think the real declaration is more clear

Reply:
idea is to hide the magic of the THREADSTATIC stuff so we can later replace it if we need to.
E.g. the var and type and number of args may change. it follows the structure we use for TDITHREADSTATIC vars. which are all decorated as macros making it easier to address them thruout the code.